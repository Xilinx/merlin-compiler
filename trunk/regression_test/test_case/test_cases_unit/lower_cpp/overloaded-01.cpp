//(C) Copyright 2016-2021 Xilinx, Inc.
//All Rights Reserved.
//
//Licensed to the Apache Software Foundation (ASF) under one
//or more contributor license agreements.  See the NOTICE file
//distributed with this work for additional information
//regarding copyright ownership.  The ASF licenses this file
//to you under the Apache License, Version 2.0 (the
//"License"); you may not use this file except in compliance
//with the License.  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//Unless required by applicable law or agreed to in writing,
//software distributed under the License is distributed on an
//"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
//KIND, either express or implied.  See the License for the
//specific language governing permissions and limitations
//under the License. (edited)
typedef struct {int a;} s_a;
typedef struct s_b {int b;};
void f() {
    for (int i = 0; i < 10; ++i) {
    }
}
void f(bool a) {
    int b;
    f();
}
void f(bool a, unsigned b) {
    f(a);
    f();
}

void f(s_a a) {
}

void f(s_b b) {
}

// Notice that since only one of the overloads are used by the kernel, mangling
// is not required.
#pragma ACCEL kernel
int main() {
    bool a;
    unsigned b;
    s_a sa;
    s_b sb;
    f(sa);
    f(sb);
    f(a, b);
}
