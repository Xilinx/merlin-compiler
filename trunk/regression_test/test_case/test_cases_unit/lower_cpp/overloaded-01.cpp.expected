typedef struct {
int a;}s_a;

struct s_b 
{
  int b;
}
;

void p_Z1fv()
{
  for (int i = 0; i < 10; ++i) {
  }
}
static void __merlin_dummy_SgFunctionDeclaration_void_f_();

void p_Z1fb(bool a)
{
  int b;
  p_Z1fv();
}
static void __merlin_dummy_SgFunctionDeclaration_void_f_boo_();

void p_Z1fbj(bool a,unsigned int b)
{
  p_Z1fb(a);
  p_Z1fv();
}
static void __merlin_dummy_SgFunctionDeclaration_void_f_boo_();

void p_Z1fs_a(s_a a)
{
}
static void __merlin_dummy_SgFunctionDeclaration_void_f_s_();

void p_Z1f3s_b(struct s_b b)
{
}
static void __merlin_dummy_SgFunctionDeclaration_void_f_str_();
// Notice that since only one of the overloads are used by the kernel, mangling
// is not required.
#pragma ACCEL kernel

int main()
{
  bool a;
  unsigned int b;
  s_a sa;
  struct s_b sb;
  p_Z1fs_a(sa);
  p_Z1f3s_b(sb);
  p_Z1fbj(a,b);
}
